scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionTypingError.scala

```diff
@@ -233,43 +257,46 @@ object SpelExpressionParseError {
       expression.substring(1, expression.length - 1)
   }
 
-  object OverloadedFunctionError extends ExpressionParseError {
+  object OverloadedFunctionError extends SpelExpressionTypingError {
     override def message: String = "Could not match any overloaded method"
   }
 
   case class ArgumentTypeError(name: String, found: Signature, possibleSignatures: NonEmptyList[Signature])
-      extends ExpressionParseError {
+      extends SpelExpressionTypingError {
 
     override def message: String =
       s"Mismatch parameter types. Found: ${found
           .display(name)}. Required: ${possibleSignatures.map(_.display(name)).toList.mkString(" or ")}"
 
   }
 
-  case class GenericFunctionError(messageInner: String) extends ExpressionParseError {
+  case class GenericFunctionError(messageInner: String) extends SpelExpressionTypingError {
     override def message: String = messageInner
   }
 
-  case object PartTypeError extends ExpressionParseError {
+  case object PartTypeError extends SpelExpressionTypingError {
     override def message: String = "Wrong part types"
   }
 
-  case class ExpressionTypeError(expected: TypingResult, found: TypingResult) extends ExpressionParseError {
+  case class ExpressionTypeError(expected: TypingResult, found: TypingResult) extends SpelExpressionTypingError {
     override def message: String = s"Bad expression type, expected: ${expected.display}, found: ${found.display}"
   }
 
-  case class ExpressionCompilationError(message: String) extends ExpressionParseError
+}
 
-  case class KeyWithLabelExpressionParsingError(keyWithLabel: String, message: String) extends ExpressionParseError {
+trait SpelExpressionParseError extends ExpressionParseError
 
-    def toProcessCompilationError(
-        nodeId: String,
-        paramName: ParameterName
-    ): ProcessCompilationError.KeyWithLabelExpressionParsingError =
-      ProcessCompilationError.KeyWithLabelExpressionParsingError(keyWithLabel, message, paramName, nodeId)
+object SpelExpressionParseError {
 
-  }
+  case class SpelExpressionUnderlyingParserError(message: String, override val details: Option[TextErrorDetails])
```

Comment:
Do we need this case class? I noticed it is used only once, in `SpelExpression.baseParse`. 

Maybe the underlying spring's class `ParseException` could be wrapped in some new implementation of `SpelExpressionTypingError`. Then we could just use `SpelExpressionTypingParseError`. And when the exception is something different than `ParseException` maybe we could just throw an exception (because IMO we should not wrap other exceptions as `SpelExpressionUnderlyingParserError`)

Reply:
We have no control over what Exception spel throwns. Runtime exceptions are not visible in the interfaces so we can't assume that the only possible exception that can be thrown is `ParseException`. I don't want to block user when we bump spel and some new type of Exception will be thrown.